oidc-client/Internal.ml

```diff
@@ -4,13 +4,15 @@ let to_string_body (res : Piaf.Response.t) = Piaf.Body.to_string res.body
 
 let read_registration ~http_client ~client_id ~(discovery : Oidc.Discover.t) =
   match discovery.registration_endpoint with
-  | Some endpoint ->
+  | Some endpoint -> (
     let open Lwt_result.Infix in
     let registration_path = endpoint |> Uri.path in
     let query = Uri.encoded_of_query [("client_id", [client_id])] in
     let uri = registration_path ^ query in
     Piaf.Client.get http_client uri >>= to_string_body >>= fun s ->
-    Oidc.Client.dynamic_of_string s |> Lwt.return
+    match Oidc.Client.dynamic_of_string s with
+    | Ok s -> Lwt_result.return s
+    | Error e -> Lwt_result.fail (`Msg e))
```

Comment:
This is wrapping any error in `` `Msg `` which is probably not what we want, we probably want `Lwt_result.fail e`.

But I also think this change is unnecessary since it's doing the same thing as before if I'm not mistaken.

Reply:
Nvm, the signature was changed further down, this is why you don't use "Add single comment"